### ISSUE TITLE
npyio.loadtxt is bytes-casting text file input, even with str dtype specified.

### ISSUE THREAD

**Issue by Panoplos** [numpy/numpy#2715]
**Environment**
- Python: Version 3.3 (python.org release) on OS X Mountain Lion
- numpy: Cloned from git master

When calling `numpy.loadtxt` on file containing strings as follows:

``` python
import numpy as np
datestxt = np.loadtxt("NYSE_dates.txt", dtype=str)
print(datetxt)
```

Where NYSE_dates.txt is simply a list of dates (could be anything really):

```
7/5/1962
7/6/1962
7/9/1962
...
12/29/2020
12/30/2020
12/31/2020
```

Output is:

```
["b'7/5/1962'" "b'7/6/1962'" "b'7/9/1962'" ..., "b'12/29/2020'"
 "b'12/30/2020'" "b'12/31/2020'"]
```

As you can see, all the strings have been bytes-casted, then stringified through `conv`, as you would get the same result from `str(str('12/31/2020').encode('latin1'))`, per `conv` & `compat.asbytes`.

After looking at the code, it appears that all strings are bytes-casted with `asbytes(...)` pretty much throughout, as for example in `split_line(...)`, so this must mean every routine in the module is broken.

**Comment by charris** [comment]
Pushing off to 1.11.

**Comment by danizen** [comment]
work-around - run iconv on the file first.

**Comment by charris** [comment]
pushing off to 1.11.

**Comment by paalge** [comment]
I see that this is being pushed forward, but I find that is is a bug that should be addressed, and a fix seems to be easily implemented.

**Comment by Queuecumber** [comment]
Pretty shocking that this hasn't been fixed for 5 years

**Comment by mdickinson** [comment]
It looks as though this is working as desired in NumPy 1.13.3 (though I'm not sure which PR fixed it). Can this issue by closed?
```
>>> import io
>>> import numpy as np
>>> f = io.StringIO("7/5/1962\n7/6/1962\n")
>>> np.loadtxt(f, dtype=str)
array(['7/5/1962', '7/6/1962'],
      dtype='<U8')
>>> np.__version__
'1.13.3'
```

**Comment by mdickinson** [comment]
Looks like this was fixed in #8349, in response to #8033.

**Comment by mattip** [comment]
Closing. Please reopen if needed.
